tenets.md

```diff
@@ -0,0 +1,433 @@
+if the danger of misuse is too high.
+This is not simply a case of cost vs. benefit.
+The downside risk of misuse SHOULD be evaluated
+independent of the value of new features.
+
+## Correct by construction
+
+### Summary
+
+Correctness can be evaluated
+at different stages of construction.
+
+1. Development
+
+   During development the code
+   as written may not satisfy the requirements.
+   It is rarely the case that
+   the code satisfies none of the requirements.
+   The problem is more often
+   that the code is only _mostly_ correct
+   as opposed to **all** correct.
+
+1. Initialization
+
+   At runtime when components are configured,
+   instantiated, or created
+   not all combinations are correct.
+
+1. Use
+
+   At runtime when the configured components
+   are used dynamically
+   during steady state operation,
+   not all inputs are correct.
+
+The AWS Encryption SDK (ESDK) is a component
+that customers use to achieve their security requirements.
+If the ESDK is not correct,
+then it is impossible for customers
+to reason about their security requirements.
+
+Given a component that operates correctly,
+customers want the configuration that is correct for them.
+The above stages also apply to the use of the ESDK.
+
+Again furniture assembly is an example.
+By providing accurate pictures
+of each step and the finished product
+it is obvious after all the parts
+have been assembled that the table is complete.
+
+It is easier to verify
+that a system is correct
+when you're creating it,
+because systems tend toward complexity
+over time.
+As systems evolve, small, seemingly minor, changes
+can have unwanted emergent behavior.
+It is unlikely that your system
+will be _less_ complicated as time goes on.
+This is true both for the ESDK
+and for our customers using the ESDK.
+
+### Motivation
+
+The ESDK's priority is to
+create interfaces that are correct by construction.
+Our priority in identifying incorrectness MUST be
+
+1. Development
+1. Initialization
+1. Use
+
+Early detection of errors
+helps us deliver correct code to customers
+but it also helps customers build
+systems that have steady state stability.
+
+Our customers MUST be able to
+clearly express their requirements
+in the construction of their calls to our APIs.
+
+As a developer is integrating the ESDK into a project
+they MUST be able to [easily](#easy-to-use) transform
+a set of requirements into a implementation
+that is obviously correct.
+
+Similarly, reviewers or maintainers
+SHOULD be able to understand
+how the implementation satisfies the requirements.
+
+Being able to easily compose the ESDK parts
+to satisfy all of your requirements is not enough.
+We expect that the complete list of requirements
+to have conflicting security needs for different types of data.
+A single component that satisfies
+multiple conflicting requirements
+may be complete and expressive,
+but it is also confusing
+and is not [hard to misuse](#hard-to-misuse).
```

Comment:
nit: I find this double (triple?) negative easy to misunderstand (not hard to misunderstand?). Maybe "easy to misuse" or "hard to use" would be better?

Reply:
taking it.  I wanted the symmetry, but I think that it is still obvious.